doc/source/reference/random/index.rst

```diff
@@ -190,7 +190,7 @@ Concepts
    :maxdepth: 1
 
    generator
-   legacy mtrand <legacy>
+   Legacy generator and functions <legacy>
```

Comment:
To me this is confusing, mtrand is its own class and does not inherit from generator

Reply:
Maybe more verbose is better: "Legacy generator (mtrand) and functions" ?

Reply:
`mtrand` is a private implementation detail, it should not be mentioned anywhere. Happy to change it if you have another suggestion, but I think this is about as accurate and clear as it gets.

Reply:
I like `Legacy Generator` or `Legacy Generator (RandomState)` which might help some users familiar with RandomState.  Not sure `and functions` feels meaningful.

Reply:
Ah, I missed the actual point here: I thought that page had `numpy.random.<function>` links, but it doesn't. That shows a bigger problem though, those functions (e.g. `np.random.seed`, `np.random.randn`) are completely missing from the docs.

that's a larger fix, so for now I'll just push a commit to address this comment, then this PR can be merged.